packages/webapp/src/components/Finances/UpdateEstimatedCropRevenue/index.jsx

```diff
@@ -57,30 +55,15 @@ function PureUpdateEstimatedCropRevenue({ system, managementPlan, onGoBack, onSu
   const ESTIMATED_ANNUAL_REVENUE = 'crop_management_plan.estimated_revenue';
 
   const disabled = !isValid;
+  const pricePerMass = watch(ESTIMATED_PRICE_PER_UNIT);
+  const annualYield = watch(ESTIMATED_ANNUAL_YIELD);
 
-  const calculateRevenue = (e) => {
-    const pricePerMass = getValues(ESTIMATED_PRICE_PER_UNIT);
-    const pricePerMassUnit = getValues(ESTIMATED_PRICE_PER_UNIT_UNIT);
-    const annualYield = getValues(ESTIMATED_ANNUAL_YIELD);
-    const annualYieldUnit = getValues(ESTIMATED_ANNUAL_YIELD_UNIT);
-    if (!pricePerMass || !annualYield) return;
-    const convertedPricePerMass = roundToTwoDecimal(
-      convert(pricePerMass).from(seedYield.databaseUnit).to(pricePerMassUnit.value),
-    );
-    const convertedAnnualYield = roundToTwoDecimal(
-      convert(annualYield).from(seedYield.databaseUnit).to(annualYieldUnit.value),
-    );
-    if (pricePerMassUnit.value === annualYieldUnit.value) {
-      const revenue = roundToTwoDecimal(convertedPricePerMass * convertedAnnualYield);
-      setValue(ESTIMATED_ANNUAL_REVENUE, revenue);
-    } else {
-      const adjustedAnnualYield = roundToTwoDecimal(
-        convert(convertedAnnualYield).from(annualYieldUnit.value).to(pricePerMassUnit.value),
-      );
-      const revenue = roundToTwoDecimal(convertedPricePerMass * adjustedAnnualYield);
-      setValue(ESTIMATED_ANNUAL_REVENUE, revenue);
+  // Calculate Revenue
+  useEffect(() => {
+    if (pricePerMass && annualYield) {
```

Comment:
When pricePerMass or annualYield is "0", Estimated annual revenue is not calculated. Was that intentional?
I think it's okay, but just confirming!

Reply:
@SayakaOno I think it was intentional haha -- why estimate zero revenue?

Reply:
Okay good!